### PR TITLE
Adds a card part that allows for oriented lists of elements

### PR DESCRIPTION
**Pull request by dtumer**
This card part orients a list of elements. The arrangement of these elements could be to the top or bottom of the view and as the view increases in size the arrangement of these elements is maintained

### CODE REVIEW

File: CardParts/src/Classes/Card Parts/CardPartOrientedView.swift [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        super.init(frame: CGRect.zero)
+        
+        // add spacer to subviews
+        let spacer = UIView()
+        spacer.translatesAutoresizingMaskIntoConstraints = false
+        spacer.setContentHuggingPriority(.defaultHigh, for: .vertical)
+        self.addSubview(spacer)
+        
+        // go through each card part and link the top margins to the bottom of the previous card part
+        for ndx in 0..<cardParts.count {
```
Comment (bharathmurs) [inline CardParts/src/Classes/Card Parts/CardPartOrientedView.swift]: `for index, cardPart in cardParts.enumerated()`

File: CardParts/src/Classes/Card Parts/CardPartOrientedView.swift [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            }
+            // if we're on the last element set the bottom of that element to the view
+            //only set bottom constraint if the orientation is bottom
+            else if ndx == cardParts.count - 1 && orientation == .bottom {
+                constraints.append(NSLayoutConstraint(item: cardPart, attribute: .bottom, relatedBy: .equal, toItem: self, attribute: .bottom, multiplier: 1, constant: 0))
+            }
+            
+            addSpacerConstraints(ndx: ndx, spacer: spacer, cardPart: cardPart)
+            self.addConstraints(constraints)
+        }
```
Comment (bharathmurs) [inline CardParts/src/Classes/Card Parts/CardPartOrientedView.swift]: `prevCardPart = cardPart`

File: CardParts/src/Classes/Card Parts/CardPartOrientedView.swift (hunk 1 above)
Comment (bharathmurs) [reply CardParts/src/Classes/Card Parts/CardPartOrientedView.swift]: Initialize prevCardPart with self (which is the superview) and at the end of the loop update prevCardPart with current.

Comment (dtumer) [reply CardParts/src/Classes/Card Parts/CardPartOrientedView.swift]: awesome fixed this in the recent commit

Comment (dtumer) [reply CardParts/src/Classes/Card Parts/CardPartOrientedView.swift]: didn't know about that function that's really cool. Good job swift

File: Example/CardParts/CardPartOrientedViewCardController.swift [hunk 3]
```diff
@@ -0,0 +1,110 @@
+//
+//  CardPartOrientedViewCardController.swift
+//  CardParts_Example
+//
+//  Created by Tumer, Deniz on 6/13/18.
+//  Copyright © 2018 CocoaPods. All rights reserved.
```
Comment (croossin) [inline Example/CardParts/CardPartOrientedViewCardController.swift]: I've been seeing it do this lately with mine as well... should change to `© 2018 Intuit`

File: CardParts/src/Classes/Card Parts/CardPartOrientedView.swift [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        let spacer = UIView()
+        spacer.translatesAutoresizingMaskIntoConstraints = false
+        spacer.setContentHuggingPriority(.defaultHigh, for: .vertical)
+        self.addSubview(spacer)
+        
+        var prevCardPart: CardPartView = self
+        
+        // go through each card part and link the top margins to the bottom of the previous card part
+        for (ndx, cardPart) in cardParts.enumerated() {
+            let cardPart = cardParts[ndx]
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartOrientedView.swift]: You don't need to fetch `cardParts[ndx]` that's why you have `for (ndx, cardPart)`.. that will give you the index and the element.

File: Example/CardParts/CardPartOrientedViewCardController.swift (hunk 3 above)
Comment (dtumer) [reply Example/CardParts/CardPartOrientedViewCardController.swift]: fixed these in most recent commit

File: CardParts/src/Classes/Card Parts/CardPartOrientedView.swift (hunk 4 above)
Comment (dtumer) [reply CardParts/src/Classes/Card Parts/CardPartOrientedView.swift]: fixed this in most recent commit

**Pull request merged**